android/anc/src/main/java/org/smartregister/fhircore/anc/ui/report/ReportViewModel.kt

```diff
@@ -43,14 +47,102 @@ class ReportViewModel(application: AncApplication, private val repository: Repor
     backPress.value = true
   }
 
-  companion object {
-    fun get(
-      owner: ViewModelStoreOwner,
-      application: AncApplication,
-      repository: ReportRepository
-    ): ReportViewModel {
-      return ViewModelProvider(owner, ReportViewModel(application, repository).createFactory())[
-        ReportViewModel::class.java]
+  fun fetchCQLLibraryData(
+    parser: IParser,
+    fhirResourceDataSource: FhirResourceDataSource,
+    libraryURL: String
+  ): LiveData<String> {
+    val libraryData = MutableLiveData<String>()
+    viewModelScope.launch(dispatcher.io()) {
+      val auxCQLLibraryData =
+        parser.encodeResourceToString(fhirResourceDataSource.loadData(libraryURL).entry[0].resource)
+      libraryData.postValue(auxCQLLibraryData)
     }
+    return libraryData
+  }
+
+  fun fetchCQLFhirHelperData(
+    parser: IParser,
+    fhirResourceDataSource: FhirResourceDataSource,
+    helperURL: String
+  ): LiveData<String> {
+    val helperData = MutableLiveData<String>()
+    viewModelScope.launch(dispatcher.io()) {
+      val auxCQLHelperData =
+        parser.encodeResourceToString(fhirResourceDataSource.loadData(helperURL).entry[0].resource)
+      helperData.postValue(auxCQLHelperData)
+    }
+    return helperData
+  }
+
+  fun fetchCQLValueSetData(
+    parser: IParser,
+    fhirResourceDataSource: FhirResourceDataSource,
+    valueSetURL: String
+  ): LiveData<String> {
+    val valueSetData = MutableLiveData<String>()
+    viewModelScope.launch(dispatcher.io()) {
+      val auxCQLValueSetData =
+        parser.encodeResourceToString(fhirResourceDataSource.loadData(valueSetURL))
+      valueSetData.postValue(auxCQLValueSetData)
+    }
+    return valueSetData
+  }
+
+  fun fetchCQLPatientData(
+    parser: IParser,
+    fhirResourceDataSource: FhirResourceDataSource,
+    patientURL: String
+  ): LiveData<String> {
+    val patientData = MutableLiveData<String>()
+    viewModelScope.launch(dispatcher.io()) {
+      val auxCQLPatientData =
+        parser.encodeResourceToString(fhirResourceDataSource.loadData(patientURL))
+      patientData.postValue(auxCQLPatientData)
+    }
+    return patientData
+  }
+
+  fun fetchCQLMeasureEvaluateLibraryAndValueSets(
+    parser: IParser,
+    fhirResourceDataSource: FhirResourceDataSource,
+    libAndValueSetURL: String,
+    measureURL: String,
+    cqlMeasureReportLibInitialString: String
+  ): LiveData<String> {
+    val valueSetData = MutableLiveData<String>()
+    val equalsIndexUrl: Int = libAndValueSetURL.indexOf("=")
+
+    val libStrAfterEquals = libAndValueSetURL.substring(libAndValueSetURL.lastIndexOf("=") + 1)
+    val libList = libStrAfterEquals.split(",").map { it.trim() }
+
+    val libURLStrBeforeEquals = libAndValueSetURL.substring(0, equalsIndexUrl) + "="
+    val initialStr = StringBuilder(cqlMeasureReportLibInitialString)
```

Comment:
is `initialStr` an appropriate name? since down below you're adding on more content, after adding that content, is the name still appropriate?

Reply:
@pld I have changed variable name to fullResourceString